nixos/modules/services/networking/nomad.nix

```diff
@@ -49,6 +49,18 @@ in
         '';
       };
 
+      settingsFiles = mkOption {
+        type = types.listOf (types.oneOf [ types.path types.str ]);
+        default = [];
+        description = ''
+          Additional settings files used to configure nomad. These files
+          will be watched for changes.
```

Comment:
This doesn't seem to be implemented.

Reply:
IIUC adding these paths to `restartTriggers` is sufficient to implement the described behavior.

Reply:
Sadly it isn't. This only works for values that are known at evaluation time, such as `config.environment.etc."nomad.json".source`. It results in a `X-Restart-Triggers` field which does not have any meaning besides changing the hash of the unit file. This means it will not read the contents of arbitrary paths at the host.

Reply:
I did consider implementing such restarts for #108411, but I left it out of scope because it was a bit complicated and it would delay progress because of that.
It can be done with systemd `path` units but it's tricky to get right. You could document the current behavior in this PR and improve it in a follow-up PR if you like.

Reply:
Got it. I found this out the hard way by adding a test that never worked, but the good news is that this module now has a test!

I'll fix up the documentation here.